base/src/main/scala/co/uproot/abandon/Ast.scala

```diff
@@ -369,4 +375,12 @@ case class Scope(entries: Seq[ASTEntry], parentOpt: Option[Scope]) extends ASTEn
         includedScopes.foreach(_.checkDupes())
     }
   }
+
+  def checkUnusedSymbols() {
+    import Console.{YELLOW, BOLD, RESET}
+    definitions.filterNot(_.isUsed)
+      .foreach(d => {
+        println(s"${YELLOW}${BOLD}symbol '${d.name}' defined in ${d.pos.filename} line: ${d.pos.pos.line} but never used${RESET}")
+      })
+  }
```

Comment:
Correct me if I'm wrong, but I think the method definitions in class Scope retrieves the parent and recursively the definitions from the included scopes in private def allLocalDefinitions:

```
private val localDefinitions = Helper.filterByType[Definition](entries)
private def allLocalDefinitions: Seq[Definition] = localDefinitions ++ includedScopes.flatMap(_.allLocalDefinitions)

  def definitions:Seq[Definition] = {
    val parentDefinitions = parentOpt.map(_.definitions).getOrElse(Nil)
    val allLocalDefs = allLocalDefinitions
    val allLocalNames = allLocalDefs.map(_.name)
    parentDefinitions.filter(d => !allLocalNames.contains(d.name)) ++ allLocalDefs
  }

```